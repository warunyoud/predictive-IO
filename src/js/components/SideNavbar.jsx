import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import MDSpinner from 'react-md-spinner';


class SideNavbar extends React.Component {

  onItemClick(index, subIndex) {
    this.props.selectIndex(index, subIndex, false);
  }

  onItemDoubleClick(index, subIndex) {
    this.props.selectIndex(index, subIndex, true);
  }

  createListNavbarItem(item, index) {
    const mainBar = [
      <div
        className={this.props.index == index && this.props.subIndex < 0 ? 'side-navbar-item-highlighted' : null }
        key={`navbarItem-${item.text}`}
        onClick={() => this.onItemClick(index, -1)}
        onDoubleClick={() => this.onItemDoubleClick(index, -1)}
        >
        <i className="material-icons" key={`navbarIcon-${item.text}`}> {item.icon} </i>
        <a key={`navbarText-${item.text}`}> {item.text} </a>
        { item.loading && <MDSpinner className='side-navbar-spinner' size='20' singleColor='#708090'/> }
      </div>
    ]

    if (this.props.index == index) {
      return _.concat(mainBar, _.map(item.subbars, (subbar, subIndex) =>
        <div
          className={this.props.subIndex == subIndex ? 'side-navbar-item-highlighted' : null }
          key={`subbarItem-${item.text}-${subbar.text}`}
          onClick={() => this.onItemClick(index, subIndex)}
          onDoubleClick={() => this.onItemDoubleClick(index, subIndex)}
        >
          <a>{subbar.text}<sup>{subbar.unsaved ? '*' : '' }</sup></a>
        </div>
      ));
    } else {
      return mainBar;
    }
  }

  render() {
    return (
      <div className='side-navbar'>
        { _.map(this.props.items, (item, index) => this.createListNavbarItem(item, index)) }
      </div>
    );
  }
}

SideNavbar.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  index: PropTypes.number.isRequired,
  subIndex: PropTypes.number.isRequired,
  selectIndex: PropTypes.func.isRequired,
}

export default SideNavbar;
