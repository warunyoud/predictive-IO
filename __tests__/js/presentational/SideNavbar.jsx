import React from 'react';
import SideNavbar from '../../../src/js/presentational/SideNavbar';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const navbarItems = [
    { icon: 'storage', text: 'Dataset', subbars: [], loading: false },
    { icon: 'layers', text: 'Train', subbars: [], loading: false },
    { icon: 'gps_fixed', text: 'Predict', subbars: [], loading: false },
    { icon: 'assessment', text: 'Evaluate', subbars: [], loading: false },
    { icon: 'code', text: 'Console', subbars: [], loading: false },
  ]
  const tree = renderer
    .create(<SideNavbar
      items={ navbarItems }
      index={ 0 }
      subIndex={ 0 }
      selectIndex={ jest.fn() }
      selectSubIndex={ jest.fn() }
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
