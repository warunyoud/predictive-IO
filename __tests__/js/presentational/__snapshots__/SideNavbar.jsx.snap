// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="side-navbar"
>
  <div
    className={null}
    onClick={[Function]}
    onDoubleClick={[Function]}
  >
    <i
      className="material-icons"
    >
       
      storage
       
    </i>
    <a>
       
      Dataset
       
    </a>
  </div>
  <div
    className={null}
    onClick={[Function]}
    onDoubleClick={[Function]}
  >
    <i
      className="material-icons"
    >
       
      layers
       
    </i>
    <a>
       
      Train
       
    </a>
  </div>
  <div
    className={null}
    onClick={[Function]}
    onDoubleClick={[Function]}
  >
    <i
      className="material-icons"
    >
       
      gps_fixed
       
    </i>
    <a>
       
      Predict
       
    </a>
  </div>
  <div
    className={null}
    onClick={[Function]}
    onDoubleClick={[Function]}
  >
    <i
      className="material-icons"
    >
       
      assessment
       
    </i>
    <a>
       
      Evaluate
       
    </a>
  </div>
  <div
    className={null}
    onClick={[Function]}
    onDoubleClick={[Function]}
  >
    <i
      className="material-icons"
    >
       
      code
       
    </i>
    <a>
       
      Console
       
    </a>
  </div>
</div>
`;
